platforms/ide/problems-api/src/main/java/org/gradle/api/problems/deprecation/DeprecateSpec.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.deprecation;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.problems.DocLink;
+
+/**
+ * Specification for building deprecations related to behavior.
+ *
+ * @since 8.14
+ */
+@Incubating
+public interface DeprecateSpec {
+
+    /**
+     * Declares a longer, potentially multi-line detailing of the deprecation.
+     * Possibly can be used to detail the reasons of the deprecation.
+     *
+     * @return the fluent builder used to call this
+     * @since 8.14
+     */
+    DeprecateSpec details(String reason);
```

Comment:
💭  I am not sure it's the right name for the method, and the slight disconnect between the method name, parameter name and the javadoc hints at that too.

Also, this is a primary method we want all users of the API to call. In that sense, `details` feels secondary and not primary.

Kotlin calls it a `message`, which I think is a better fit: https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-deprecated/ Very much like a message for the exception, and serves all the same purposes that are stated in the javadoc.